src/vs/workbench/contrib/chat/browser/chatWidget.ts

```diff
@@ -1046,6 +1065,29 @@ export class ChatWidget extends Disposable implements IChatWidget {
 				this.telemetryService.publicLog2<ChatEditingWorkingSetEvent, ChatEditingWorkingSetClassification>('chatEditing/workingSetSize', { originalSize: this.inputPart.attemptedWorkingSetEntriesCount, actualSize: uniqueWorkingSetEntries.size });
 			}
 
+			// factor in nested references of dynamic variables into the implicit attached context
+			const variableModel = this.getContrib<ChatDynamicVariableModel>(ChatDynamicVariableModel.ID);
```

Comment:
If the ChatWidget calls to a contrib directly, then it's no longer a "contrib". The point of that model is to have something that can implement a feature on top of an object using its public interface, but is totally decoupled from the object.

Reply:
So either this should no longer be a 'contrib' or something else needs to change, I think I need to understand the overall thing better to decide.

Reply:
I needed to get to the `variables` list to extend the attached context with the resolved nested file references below.
Maybe there is a better place of doing this, or a better place to add the nested file references to? 🤔

Reply:
Moved to `ChatInputPart.getAttachedAndImplicitContext()` since we add the nested file references as an implicit context atm 🚀